src/game.cpp

```diff
@@ -5288,11 +5313,21 @@ void Game::playerAcceptMarketOffer(uint32_t playerId, uint32_t timestamp, uint16
 			buyerPlayer->onReceiveMail();
 		}
 	} else {
-		if (totalPrice > player->bankBalance) {
+		if (totalPrice > (player->getMoney() + player->bankBalance)) {
 			return;
 		}
 
-		player->bankBalance -= totalPrice;
+		uint64_t playerMoney = player->getMoney();
+		if (totalPrice <= playerMoney) {
+			removeMoney(player, totalPrice);
+		} else if (totalPrice <= (playerMoney + player->bankBalance)) {
+			if (playerMoney > 0) {
+				removeMoney(player, playerMoney);
+				player->bankBalance -= (totalPrice - playerMoney);
+			} else {
+				player->bankBalance -= totalPrice;
+			}
+		}
```

Comment:
```suggestion
		const auto debitCash = std::min(player->getMoney(), totalPrice);
		const auto debitBank = totalPrice - debitCash;
		removeMoney(player, debitCash);
		player->bankBalance -= debitBank ;
```

Wouldn't this be simpler and achieve same purpose?

Reply:
Yes! cool code